vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/Grid.java

```diff
@@ -3152,6 +3170,12 @@ public void setMultiSort(boolean multiSort, MultiSortPriority priority) {
                 "Multi-sort priority must not be null");
         setMultiSort(multiSort);
 
+        if (priority != defaultMultiSortPriority) {
```

Comment:
This check may be problematic when changing the priority dynamically (checking the value agains the default one, but not the effective one)

Reply:
Thanks, changed to always update the priority to avoid such case.